code/game/dna/genes/vg_powers.dm

```diff
@@ -37,70 +37,101 @@
 	if(new_eyes)
 		M.r_eyes = hex2num(copytext(new_eyes, 2, 4))
 		M.g_eyes = hex2num(copytext(new_eyes, 4, 6))
 		M.b_eyes = hex2num(copytext(new_eyes, 6, 8))
-
-	var/new_tone = input("Please select skin tone level: 1-220 (1=albino, 35=caucasian, 150=black, 220='very' black)", "Character Generation", "[35-M.s_tone]") as null|text
-
-	if (!new_tone)
-		new_tone = 35
-	M.s_tone = max(min(round(text2num(new_tone)), 220), 1)
-	M.s_tone =  -M.s_tone + 35
+		M.change_eye_color(M.r_eyes, M.g_eyes, M.b_eyes)
 
 	// hair
-	var/list/all_hairs = subtypesof(/datum/sprite_accessory/hair)
-	var/list/hairs = list()
-
-	// loop through potential hairs
-	for(var/x in all_hairs)
-		var/datum/sprite_accessory/hair/H = new x // create new hair datum based on type x
-		hairs.Add(H.name) // add hair name to hairs
-		qdel(H) // delete the hair after it's all done
-
-	var/new_style = input("Please select hair style", "Character Generation",M.h_style)  as null|anything in hairs
+	var/list/valid_hairstyles = M.generate_valid_hairstyles()
+	var/new_style = input("Please select hair style", "Character Generation", head_organ.h_style) as null|anything in valid_hairstyles
 
 	// if new style selected (not cancel)
 	if (new_style)
-		M.h_style = new_style
-
-	// facial hair
-	var/list/all_fhairs = subtypesof(/datum/sprite_accessory/facial_hair)
-	var/list/fhairs = list()
+		head_organ.h_style = new_style
 
-	for(var/x in all_fhairs)
-		var/datum/sprite_accessory/facial_hair/H = new x
-		fhairs.Add(H.name)
-		qdel(H)
+	var/new_hair = input("Please select hair color.", "Character Generation", rgb(head_organ.r_hair, head_organ.g_hair, head_organ.b_hair)) as null|color
+	if(new_hair)
+		head_organ.r_hair = hex2num(copytext(new_hair, 2, 4))
+		head_organ.g_hair = hex2num(copytext(new_hair, 4, 6))
+		head_organ.b_hair = hex2num(copytext(new_hair, 6, 8))
 
-	new_style = input("Please select facial style", "Character Generation",M.f_style)  as null|anything in fhairs
+	// facial hair
+	var/list/valid_facial_hairstyles = M.generate_valid_facial_hairstyles()
+	new_style = input("Please select facial style", "Character Generation", head_organ.f_style) as null|anything in valid_facial_hairstyles
 
 	if(new_style)
-		M.f_style = new_style
+		head_organ.f_style = new_style
 
-	var/new_gender = alert(usr, "Please select gender.", "Character Generation", "Male", "Female")
-	if (new_gender)
-		if(new_gender == "Male")
-			M.change_gender(MALE)
-		else
-			M.change_gender(FEMALE)
+	var/new_facial = input("Please select facial hair color.", "Character Generation", rgb(head_organ.r_facial, head_organ.g_facial, head_organ.b_facial)) as null|color
+	if(new_facial)
+		head_organ.r_facial = hex2num(copytext(new_facial, 2, 4))
+		head_organ.g_facial = hex2num(copytext(new_facial, 4, 6))
+		head_organ.b_facial = hex2num(copytext(new_facial, 6, 8))
+
+	//Head accessory.
+	if(head_organ.species.bodyflags & HAS_HEAD_ACCESSORY)
```

Comment:
Means that if you aren't a species that has head accessories, this option won't pop up when you use the morph ability.